pkg/registry/apis/provisioning/plog/plog.go

```diff
@@ -0,0 +1,38 @@
+package plog
+
+import (
+	"context"
+	"log/slog"
+)
+
+type key int
```

Comment:
If we don't want to import a new package which already provides the utils to use a context logger, I would do the following: 

- Create a package `ctxslog`
- Default to `slog.Default()` if a logger is not found in the context. 
- Keep the `ctxslog` as simply as possible without creating our own logger. Just using `slog.Logger` in the signatures. 
- Add `With` `FromContext` and `ToContext` function which are what we need right now. 


See what this package offers as a reference: https://pkg.go.dev/github.com/veqryn/slog-context#section-documentation

Reply:
WDYT about the slogctx I pushed now? Still gotta start using it, but this is the API I reckon is good enough.

Reply:
Much aligned to what I had in mind :) See the comments below about making the functions single-purpose :)